og_context/og_context.module

```diff
@@ -436,26 +439,26 @@ function og_context_determine_context($group_type, $item = NULL, $check_access =
       $invoke = TRUE;
     }
 
-    $gids = array();
-
     if ($invoke && ($contexts = call_user_func($provider['callback'])) && !empty($contexts[$group_type])) {
-      // Check if one of the group IDs already exists in the session, and if
-      // so use it.
+      // Check if one of the group IDs already exists in the session,
+      // and if so give it priority over the other IDs.
       $gids = $contexts[$group_type];
-      if (!empty($_SESSION['og_context']['group_type']) && $_SESSION['og_context']['group_type'] == $group_type && in_array($_SESSION['og_context']['gid'], $gids)) {
-        $check_gid = $_SESSION['og_context']['gid'];
-      }
-      else {
-        // Grab the first group ID.
-        $check_gid = reset($gids);
+      if ($sessiontype === $group_type && in_array($sessiongid, $gids) && $gids[0] != $sessiongid) {
+        array_unshift($gids, $_SESSION['og_context']['gid']);
```

Comment:
Needs a comment what we do here.

Reply:
The comment explaining this is actually present but not directly above this code. I will move it around a bit.